test/e2e/specs/authentication/authentication__google.ts

```diff
@@ -173,43 +152,19 @@ describe( 'Authentication: Google', function () {
 		} );
 
 		it( 'Enter 2FA challenge if required - Challenge 1', async function () {
-			// Add a handler that will detect whether the popup page
-			// has closed.
-			// The popup being closed/dismissed signifies the login
-			// process was successful.
-			// This handler, like all other "page.on"-type handlers
-			// will persist for the lifetime of the thread.
-			// In other words, this handler will remain active until
-			// the Jest process is terminated.
-			googlePopupPage.on( 'close', () => {
-				windowClosed = true;
-			} );
-
-			try {
-				// Separately from above, create a promise that will
-				// resolve once the popup page closes.
-				// This promise is then awaited on to see if it will
-				// resolve after the TOTP code is entered.
-				const googlePopupPageClosePromise = googlePopupPage.waitForEvent( 'close' );
-
-				totpClient = new TOTPClient( credentials.totpKey as string );
-				code = totpClient.getToken();
-
-				await googleLoginPage.enter2FACode( code );
-				await googleLoginPage.clickButton( 'Next' );
-
-				// If the promise resolves (ie. popup closes) then
-				// the TOTP was accepted AND the login is successful.
-				await googlePopupPageClosePromise;
-			} catch {
-				console.log( 'Expecting second 2FA challenge.' );
-				// noop
-				//
-				// If `googlePopupPageClosePromise` rejects, it means
-				// the popup did not close.
-				// This in turn means the TOTP was rejected, or
-				// the unique circumstance we see (below) has come into play.
+			// Wait until the TOTP code generated by the client
+			// changes, meaning the 30s window has rolled over.
+			while ( totpClient.getToken() === code ) {
```

Comment:
Since we use the same seed and the client is time-based, we need to have the same check for any subsequent `getToken()` calls after the first call. I tried creating a new instance (as it was before), and it doesn't work. The same code is returned if the token hasn't expired yet. The only way to improve the test would be to dive into time traveling (skip the time by 30 seconds in the future), and I think we should explore that as a follow-up to speed up this test, but I guess this is good enough to fix the test for now.